libdebug/ptrace/ptrace_interface.py

```diff
@@ -250,10 +248,22 @@ def cont(self: PtraceInterface) -> None:
         else:
             self._global_state.handle_syscall_enabled = False
 
-        result = self.lib_trace.cont_all_and_set_bps(
-            self._global_state,
-            self.process_id,
-        )
+    def cont(self: PtraceInterface, thread: ThreadContext | None) -> None:
+        """Continues the execution of the process."""
+        self._prepare_for_cont()
```

Comment:
We are setting the signals for every thread, even if we are continuing only once (and therefore delivering just the signal for the continued thread). It’s not the cleanest choice, but theoretically, there shouldn't be any side effects. I suppose.